### PR TITLE
FeedbackRankQuestionUiTest.json: Fix data for Tests #8794

### LINKED ISSUE #8794: FeedbackRankQuestionUiTest.json: Fix data for Tests
Problem spotted in #8710.

The `FeedbackRankQuestionUiTest.json` has a problem in the data used for the test case.

These are the four students which are in the data. 
**Alice and Benny** belong to _Team1_ and **Charlie and Danny** belong to _Team2_.
https://github.com/TEAMMATES/teammates/blob/6157ad411338974355998471d034523472c2d191/src/test/resources/data/FeedbackRankQuestionUiTest.json#L100-L136

However for responses to question 3 which has `recipientType `of the question is `OWN_TEAM_MEMBERS_INCLUDING_SELF`, 
https://github.com/TEAMMATES/teammates/blob/6157ad411338974355998471d034523472c2d191/src/test/resources/data/FeedbackRankQuestionUiTest.json#L617-L643

There exists a response from Benny to Charlie[who are in different team]
https://github.com/TEAMMATES/teammates/blob/6157ad411338974355998471d034523472c2d191/src/test/resources/data/FeedbackRankQuestionUiTest.json#L976-L988

We need to fix the data and change test cases accordingly to avoid breaking anything and have correct data

### PR DESCRIPTION
**Pull request by mmoanis**
Fixes #8794 
There are 7 answers for the feedback rank question number 3 in the test data, but there are 3 scenarios that are incorrect:
- Benny (team 1) gives feedback to Charlie (team 2)
- Alice (team 1) gives feedback to Charlie (team 2)
- Charlie (team 2) gives feedback to Alice (team 1)

I corrected these scenarios and updated the baseline HTML pages for the test case that uses this JSON.

Please also review the section number in the feedback, I think they are incorrect too?

### CODE REVIEW

**Comment by mmoanis** [comment]
cc @tshradheya

**Comment by tshradheya** [comment]
> Please also review the section number in the feedback, I think they are incorrect too?

Yeah agree. I think that error is only in the feedbackresponsedata. 

Cause here all students belong to Section 1
https://github.com/TEAMMATES/teammates/blob/6157ad411338974355998471d034523472c2d191/src/test/resources/data/FeedbackRankQuestionUiTest.json#L100-L136

You can correct that as well. It should not affect any other part

**Comment by mmoanis** [comment]
Done. Ready for review.

**Comment by tshradheya** [comment]
I just noticed that the section number is wrong for many such feedbackResponseData.

Let's correct them everywhere in this json file since it isnt causing anything else to break

**Comment by mmoanis** [comment]
Cool. Good Luck!

**Comment by mmoanis** [comment]
oops, I 'grep'ed for "Section 2" only.
Thanks for the catch, ready to go.

**Comment by tshradheya** [comment]
@mmoanis Please update your branch with the latest changes from Master.

I am expecting 1 test to fail, which you should be able to fix by running in GodMode

**Comment by tshradheya** [comment]
Verified that after making the chnages, the HTML now shows correct data and tests are now more meaningful. 

@LiHaoTan @Crphang Can either of you please do the second round of review?(After the PR is synced with lastest changes in master

**Comment by tshradheya** [comment]
@TEAMMATES/seniordevs Can someone have a quick look at this? It's blocking #8170 so it would be great if we can fix this soon

File: src/test/resources/data/FeedbackRankQuestionUiTest.json [hunk 1]
```diff
@@ -1109,7 +1109,7 @@
       "responseMetaData": {
         "value": "{\"answer\":2,\"questionType\":\"RANK_RECIPIENTS\"}"
       },
-      "giverSection": "None",
+      "giverSection": "Section 1",
```
Comment (LiHaoTan) [inline src/test/resources/data/FeedbackRankQuestionUiTest.json]: Self feedback rank recipient questions are no longer valid (#8218), but this is not enforced in the backend, we need to create a corresponding issue for it.

Comment (tshradheya) [reply src/test/resources/data/FeedbackRankQuestionUiTest.json]: @mmoanis You can either go ahead and remove this question from test data or see if it is possible to modify the question in an easy way

**Comment by tshradheya** [comment]
@mmoanis Can you make the 1 requested change at the earliest so this could be merged soon. Its blocking #8710

**Comment by mmoanis** [comment]
@tshradheya sorry I will not be able to continue in this for the time being. closing

**Pull request closed without merging**